modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/IntrospectionUtils.java

```diff
@@ -183,33 +182,13 @@ public final class IntrospectionUtils {
   private static final Logger LOGGER = getLogger(IntrospectionUtils.class);
 
   static {
-    setWeakHashCaches();
+    // A shade of the spring-core library is the cause for this additional cleanup.
```

Comment:
Please add a comment referencing this in the pom.xml that configures the shade plugin, so we don't forget to remove this in that case.